lib/queue/events.ts

```diff
@@ -260,34 +131,32 @@ export const wait = async (
 		stream.once('closed', () => {
 			stream.removeAllListeners();
 			logger.info(logContext, 'Closing wait stream', {
-				slug,
+				id: actionRequestId,
 			});
 
 			resolve(result);
 		});
 
-		// Don't perform a "get by slug" if we already have a match.
+		// Don't attempt a direct query if we already have a match.
 		if (result) {
 			return;
 		}
 
 		kernel
-			.getContractBySlug(
-				logContext,
-				session,
-				`${slug}@${EXECUTION_EVENT_VERSION}`,
-			)
-			.then((contract) => {
-				if (!contract) {
+			.query(logContext, session, schema, {
+				limit: 1,
+			})
+			.then((contracts) => {
+				if (!contracts[0]) {
 					return;
 				}
 
-				logger.info(logContext, 'Found results on first slug query', {
-					slug: contract.slug,
-					data: Object.keys(contract.data),
+				logger.info(logContext, 'Found results on query', {
+					id: contracts[0].id,
+					data: Object.keys(contracts[0].data),
 				});
 
-				result = result || (contract as ExecuteContract);
+				result = result || (contracts[0] as ActionRequestContract);
```

Comment:
This could use the `||=` assignment instead as it's a little clearer on the intent of a conditional assignment
```suggestion
				result ||= contracts[0] as ActionRequestContract;
```
vs the current which can also be used as `a = x || y` and have different behavior, alternatively using `??` might actually be more accurate to the actual intent (I tend to see `||` used in place of `??` in a bunch of places in codebases that predate `??`)
```suggestion
				result ??= contracts[0] as ActionRequestContract;
```

Reply:
@Page- I think you're right, this is a line that's been there since before I even joined the company, my guess is that its from the age before `??`. Updated to `??=`.